src/pybind/mgr/dashboard/controllers/nvmeof.py

```diff
@@ -38,12 +40,42 @@ def list(self):
     @APIDoc("NVMe-oF Subsystem Management API", "NVMe-oF Subsystem")
     class NVMeoFSubsystem(RESTController):
         @EndpointDoc("List all NVMeoF subsystems")
-        @map_collection(model.Subsystem, pick="subsystems")
-        @handle_nvmeof_error
         def list(self):
-            return NVMeoFClient().stub.list_subsystems(
+            response = []
+            # Get list of subystems
+            subsystems = NVMeoFClient().stub.list_subsystems(
                 NVMeoFClient.pb2.list_subsystems_req()
             )
+            if subsystems.status != 0:
+                raise DashboardException(
+                    msg=response.error_message,
+                    code=response.status,
+                    http_status_code=NVMeoFClient().NVMeoFError2HTTP.get(response.status, 400),
+                    component="nvmeof",
+                )
+            # Create a new response with initiators information appended
+            for subs in subsystems.subsystems:
+                res = {
+                    'nqn': subs.nqn,
+                    'enable_ha': subs.enable_ha,
+                    'serial_number': subs.serial_number,
+                    'model_number': subs.model_number,
+                    'min_cntlid': subs.min_cntlid,
+                    'max_cntlid': subs.max_cntlid,
+                    'namespace_count': subs.namespace_count,
+                    'max_namespaces': subs.max_namespaces,
+                    'subtype': subs.subtype,
+                    'initiators': [],
+                }
+                # Get list of initiators present on the subsystem
+                hosts = NVMeoFHost().list(subs.nqn)
+                # Decode the bytes object to json
+                json_str = hosts.decode('utf-8')
+                json_obj = json.loads(json_str)
+                # Appending initiators information
+                res['initiators'] = json_obj
+                response.append(res)
+            return response
```

Comment:
Why changing the existing approach? It basically used decorators to programatically deal with:
* mapping gRPC entities to REST/JSON/NamedTuples,
* Error handling.

Reply:
Since this is a custom response therefore it was failing for me and I tred doing something like { subsystems: response, status: 0, status_message: "Success" } but it did not work.
If you have any suggestions I could revert then, thanks!